helm-charts/basehub/values.yaml

```diff
@@ -441,3 +441,74 @@ jupyterhub:
 
         if get_config("custom.docs_service.enabled"):
             c.JupyterHub.services.append({"name": "docs", "url": "http://docs-service"})
+      09-gh-teams: |
+        from textwrap import dedent
+        from tornado import gen, web
+
+        # Make a copy of the original profile_list, as that is the data we will work with
+        original_profile_list = c.KubeSpawner.profile_list
+
+        # This has to be a gen.coroutine, not async def! Kubespawner uses gen.maybe_future to
+        # run this, and that only seems to recognize tornado coroutines, not async functions!
+        # We can convert this to async def once that has been fixed upstream.
+        @gen.coroutine
+        def custom_profile_list(spawner):
+            """
+            Dynamically set allowed list of user profiles based on GitHub teams user is part of.
+
+            Adds a 'allowed_teams' key to profile_list, with a list of GitHub teams (of the form
+            org-name:team-name) for which the profile is made available.
+
+            If the user isn't part of any team whose membership grants them access to even a single
+            profile, they aren't allowed to start any servers.
+            """
+
+            # If populate_teams_in_auth_state is not set, github teams are not fetched
+            # So we just don't do any of this filtering, and let anyone into everything
+            if spawner.authenticator.populate_teams_in_auth_state == False:
+              return original_profile_list
+
+            auth_state = yield spawner.user.get_auth_state()
+
+            if not auth_state or "teams" not in auth_state:
+              if spawner.user.name == 'deployment-service-check':
+                # For our hub deployer health checker, ignore all this logic
+                print("Ignoring allowed_teams check for deployment-service-check")
+                return original_profile_list
```

Comment:
Now that we own https://github.com/deployment-service-check/ maybe we should add that account to our GitHub org and tech team and remove this logic? I wonder if we can close the security backdoor by treating `deployment-service-check` as an actual member of our team?

Reply:
@sgibson91 we couldo that if we can make that user authenticate using the oauth flow - although last i looked that was pretty difficult :(

Reply:
Ah okay